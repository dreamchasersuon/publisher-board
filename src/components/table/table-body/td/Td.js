import React from 'react';
import PropTypes from 'prop-types';
import './Td.css';

export function calcDaysFromRegistration(date) {
  const dayInMs = 24 * 60 * 60 * 1000;
  const now = new Date();
  const registered = new Date(date);
  return Math.round(Math.abs((now - registered) / dayInMs));
}

export default function Td({ isDate, value }) {
  if (isDate) {
    return (
      <td>
        <div className="TableBody-cell">{calcDaysFromRegistration(value)}d</div>
      </td>
    );
  } else if (typeof value === 'number') {
    return (
      <td>
        <div className="TableBody-cell">{value}</div>
      </td>
    );
  } else {
    return (
      <td>
        <div className="TableBody-cell">{value || 'â€”'}</div>
      </td>
    );
  }
}

Td.propTypes = {
  isDate: PropTypes.bool,
  value: PropTypes.any,
};
