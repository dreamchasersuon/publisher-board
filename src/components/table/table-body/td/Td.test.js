import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import Td, { calcDaysFromRegistration } from './Td';

describe('renders with date or with simple value, calc date before', () => {
  test('renders table body cell with date or simple value', () => {
    expect(
      snapshotDiff(
        <Td isDate value="2019-01-30T10:24:53+00:00" />,
        <Td value="123442" />,
      ),
    ).toMatchSnapshot();
  });
  test('renders table body cell with date in days before', () => {
    const dateUTC = '2019-01-30T10:24:53+00:00';
    const daysBefore = calcDaysFromRegistration(dateUTC);
    const testRenderer = renderer.create(<Td isDate value={dateUTC} />);

    const testInstance = testRenderer.root;

    expect(
      testInstance.findByProps({ className: 'TableBody-cell' }).children,
    ).toEqual([`${daysBefore}`, 'd']);
  });
});
