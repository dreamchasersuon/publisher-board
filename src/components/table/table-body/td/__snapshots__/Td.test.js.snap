// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with date or with simple value, calc date before renders table body cell with date or simple value 1`] = `
"Snapshot Diff:
- <Td isDate={true} value=\\"2019-01-30T10:24:53+00:00\\" />
+ <Td value=\\"123442\\" />

  <td>
    <div
      className=\\"TableBody-cell\\"
    >
-     391
-     d
+     123442
    </div>
  </td>"
`;
